File: android_fragment/app/src/main/java/com/support/generator/di/components/AppComponent.java
package com.support.generator.di.components;

import com.support.generator.di.App;
import com.support.generator.di.factories.ActivitiesInjectorFactories;
import com.support.generator.di.factories.FragmentsInjectorFactories;
import com.support.generator.di.modules.AppModule;

import javax.inject.Singleton;

import dagger.BindsInstance;
import dagger.Component;
import dagger.android.support.AndroidSupportInjectionModule;

/**
 * @author: leellun
 * @data: 2018/12/20.
 */
@Singleton
@Component(modules = {
        AndroidSupportInjectionModule.class,
        ActivitiesInjectorFactories.class,
        FragmentsInjectorFactories.class,
        AppModule.class
})
public interface AppComponent {

  @Component.Builder
  interface Builder {
    @BindsInstance
    public Builder application(App application);

    public AppComponent build();
  }

  public void inject(App app);

}

File: example/app/src/main/java/gauges/mobile/github/com/mvpapplication/module/di/components/AppComponent.kt
package gauges.mobile.github.com.mvpapplication.module.di.components

import gauges.mobile.github.com.mvpapplication.module.di.App
import gauges.mobile.github.com.mvpapplication.module.di.factories.ActivitiesInjectorFactories
import gauges.mobile.github.com.mvpapplication.module.di.factories.FragmentsInjectorFactories
import gauges.mobile.github.com.mvpapplication.module.di.modules.AppModule
import dagger.BindsInstance
import dagger.Component
import dagger.android.support.AndroidSupportInjectionModule
import javax.inject.Singleton

/**
 * @author: leellun
 * @data: 2018/12/19.
 *
 */
@Singleton
@Component(modules = arrayOf(
        AndroidSupportInjectionModule::class,
        ActivitiesInjectorFactories::class,
        FragmentsInjectorFactories::class,
        AppModule::class
))
interface AppComponent {

    @Component.Builder
    interface Builder {
        @BindsInstance
        fun application(application: App): Builder

        fun build(): AppComponent
    }

    fun inject(app: App)

}